.github/workflows/test-upgrade.yaml

```diff
@@ -0,0 +1,135 @@
+        uses: azure/setup-kubectl@v4
+        with:
+          version: v1.25.3
+      - name: Mask cluster token
+        if: ${{ github.event_name == 'workflow_dispatch' && inputs.clusterToken != '' }}
+        run: |
+          CLUSTER_TOKEN=$(jq -r '.inputs.clusterToken' $GITHUB_EVENT_PATH)
+          echo ::add-mask::$CLUSTER_TOKEN
+          echo CLUSTER_TOKEN=$CLUSTER_TOKEN >> $GITHUB_ENV
+      - name: Setup kubectl for remote cluster
+        if: ${{ github.event_name == 'workflow_dispatch' && inputs.clusterServer != '' }}
+        run: |
+          kubectl config set-credentials ${{ env.K8S_USER }}/${{ env.CLUSTER_NAME }} --token ${{ env.CLUSTER_TOKEN }}
+          kubectl config set-cluster ${{ env.CLUSTER_NAME }} --insecure-skip-tls-verify=true --server=${{ inputs.clusterServer }}
+          kubectl config set-context ${{ inputs.kuadrantNamespace }}/${{ env.CLUSTER_NAME }}/${{ env.K8S_USER }} --user=${{ env.K8S_USER }}/${{ env.CLUSTER_NAME }} --namespace=${{ inputs.kuadrantNamespace }} --cluster=${{ env.CLUSTER_NAME }}
+          kubectl config use-context ${{ inputs.kuadrantNamespace }}/${{ env.CLUSTER_NAME }}/${{ env.K8S_USER }}
+      ## makes sure cluster is up and running
+      - run: |
+          kubectl cluster-info
+          kubectl get nodes
+      - name: Deploy pre-requisites on local Kind cluster
+        if: ${{ github.event_name == 'workflow_dispatch' && inputs.clusterServer == '' }}
+        run: |
+          make install-metallb
+          make install-cert-manager
+          make envoy-gateway-install
+          make deploy-eg-gateway
+      - name: Install helm and add Kuadrant repo
+        run: |
+          make helm
+          make helm-add-kuadrant-repo
+          bin/helm search repo kuadrant
+      - name: Install Kuadrant ${{ inputs.kuadrantStartVersion }}
+        run: |
+          bin/helm install kuadrant kuadrant/kuadrant-operator \
+          --wait \
+          --timeout 3m0s \
+          --version ${{ inputs.kuadrantStartVersion }} \
+          --namespace ${{ inputs.kuadrantNamespace }} \
+          --create-namespace
+      - name: Deploy Kuadrant
+        run: |
+          kubectl -n ${{ inputs.kuadrantNamespace }} apply -f - <<EOF
+          apiVersion: kuadrant.io/v1beta1
+          kind: Kuadrant
+          metadata:
+            name: kuadrant
+          spec: {}
+          EOF
+      - name: Verify Kuadrant installation
+        run: |
+          kubectl wait --timeout=300s --for=condition=Ready kuadrant kuadrant -n ${{ inputs.kuadrantNamespace }}
+          echo "kuadrant installation ✅"
+      - name: Build local chart dependencies for version ${{ steps.upgrade-version.outputs.version }}
+        run: |
+          make helm-dependency-build
+      - name: Start upgrade to kuadrant ${{ steps.upgrade-version.outputs.version }}
+        run: |
+          bin/helm upgrade kuadrant charts/kuadrant-operator \
```

Comment:
we are not gonna be upgrading to pre-releases charts, are we?

Reply:
because if we are, we might need to pass the `--devel` flag

Reply:
Upgrade to pre-release charts can be tested.

The target (version being upgraded to) chart is determined by the git reference where you run the tests. Thus, the chart is available locally in the filesystem, hence the path `charts/kuadrant-operator`.